rxbilling/src/main/java/com/gen/rxbilling/client/RxBilling.kt

```diff
@@ -66,29 +64,30 @@ class RxBillingImpl(billingFactory: BillingClientFactory)
         }
     }
 
-    override fun getPurchases(): Single<List<Purchase>> {
-        return getBoughtItems(BillingClient.SkuType.INAPP)
-    }
-
-    override fun getSubscriptions(): Single<List<Purchase>> {
-        return getBoughtItems(BillingClient.SkuType.SUBS)
-    }
-
-    override fun getPurchaseHistory(): Single<List<Purchase>> {
-        return getHistory(BillingClient.SkuType.INAPP)
+    override fun getPurchases(skuType: String): Single<List<Purchase>> {
+        return getBoughtItems(skuType)
     }
 
-    override fun getSubscriptionHistory(): Single<List<Purchase>> {
-        return getHistory(BillingClient.SkuType.SUBS)
+    override fun getPurchaseHistory(skuType: String): Single<List<PurchaseHistoryRecord>> {
+        return getHistory(skuType)
     }
 
-    override fun getPurchaseSkuDetails(ids: List<String>): Single<List<SkuDetails>> {
-        return getSkuDetails(ids, BillingClient.SkuType.INAPP)
-    }
-
-
-    override fun getSubscriptionSkuDetails(ids: List<String>): Single<List<SkuDetails>> {
-        return getSkuDetails(ids, BillingClient.SkuType.SUBS)
+    override fun getSkuDetails(params: SkuDetailsParams): Single<List<SkuDetails>> {
+        return connectionFlowable
+                .flatMap { client ->
+                    Flowable.create<List<SkuDetails>>({
```

Comment:
Is there a reason to creating `Flowable` and converting it to completable via `firstOrError()` instead of just using `Completable.create{...}` here?

Reply:
But, as far as i know, we need to receive the list of SkuDetails by using this method, while the `Completable` value would emit the completion event only...

Reply:
Changed Flowable to Single.
But still need to call firstOrError() as chain starts from connection Flowable.